### PR TITLE
Add crypto to log

### PR DESCRIPTION
**Pull request by Hipst0man**

### CODE REVIEW

File: CPPLogger/CPPLogger.h [hunk 1]
```diff
@@ -48,6 +50,9 @@ class Logger final : public ILogger
 	IP7_Trace*					m_trace;
 	std::string					m_func_name;
 	std::string					m_file_name;
+	SymmetricCrypto				m_crypto;
+	std::string					m_file_dir;
+	char						buf[80];
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.h]: Change to m_buf.
And 80 is magic number.

File: LoggerUI/LogUi.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class LogUi : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    LogUi(QWidget* parent = Q_NULLPTR);
+    ~LogUi();
+
+private:
+    Ui::LogUiClass* ui;
```
Comment (CharlesLyleLaRue) [inline LoggerUI/LogUi.h]: Change to m_ui.

File: LoggerUI/LogUi.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    delete ui;
+}
+
+void LogUi::GetContent(std::ifstream& stream)
+{
+    std::vector<unsigned char> vec, dec_vec;
+    SymmetricCrypto m_crypto;
+    QString dec_line;
+
+    char ch = '\0';
```
Comment (CharlesLyleLaRue) [inline LoggerUI/LogUi.cpp]: It would be better to change null terminator to const variable.

File: LoggerUI/LogUi.cpp [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        if (stream) vec.push_back(ch);
+    }
+
+    auto vec_size = m_crypto.Decrypt(vec, dec_vec);
+    dec_vec.shrink_to_fit();
+
+    auto all_rows = 0;
+    for (auto& el : dec_vec)
+    {
+        if (el == '\n') all_rows++;
```
Comment (CharlesLyleLaRue) [inline LoggerUI/LogUi.cpp]: Make new line escape sequence as const variable.

File: LoggerUI/LogUi.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <QtWidgets/QMainWindow>
+#include "ui_LogUi.h"
```
Comment (CharlesLyleLaRue) [inline LoggerUI/LogUi.h:4]: Put an include of your source files to the bottom of include section =)

File: CPPLogger/CPPLogger.h [hunk 6]
```diff
@@ -38,7 +40,7 @@ class Logger final : public ILogger
 	void set_filter_level(unsigned int level) override;
 
 private:
-	Logger();	
+	Logger();
 	Logger(const Logger&) = default;
```
Comment (OksanaBisyk-TomTom) [inline CPPLogger/CPPLogger.h]: I think you need to write copy assignment operator here, too.

File: LoggerUI/LogUi.cpp [hunk 7]
```diff
@@ -0,0 +1,184 @@
+// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
+#include "LogUi.h"
+#include "qfiledialog.h"
+#include "qtextbrowser.h"
+#include <fstream>
```
Comment (OksanaBisyk-TomTom) [inline LoggerUI/LogUi.cpp:7]: Standard library includes should go first.

**Review by CharlesLyleLaRue — APPROVED** [review]
The time has come, Yaroslaaaaaaav!

**Review by OksanaBisyk-TomTom — APPROVED** [review]

**Pull request merged**